asb-native/src/main/java/org/ballerinax/asb/util/ASBUtils.java

```diff
@@ -222,4 +248,164 @@ public static AmqpRetryOptions getRetryOptions(BMap<BString, Object> retryConfig
                 .setTryTimeout(Duration.ofSeconds(tryTimeout.intValue()))
                 .setMode(AmqpRetryMode.valueOf(retryMode));
     }
+
+    public static RecordType getRecordType(BTypedesc bTypedesc) {
+        RecordType recordType;
+        if (bTypedesc.getDescribingType().isReadOnly()) {
+            recordType = (RecordType) getReferredType(((IntersectionType) getReferredType(
+                    bTypedesc.getDescribingType())).getConstituentTypes().get(0));
+        } else {
+            recordType = (RecordType) getReferredType(bTypedesc.getDescribingType());
+        }
+        return recordType;
+    }
+
+    /**
+     * Converts AMPQ Body value to Java objects.
+     *
+     * @param amqpValue AMQP Value type object
+     */
+    public static Object convertAMQPToJava(String messageId, Object amqpValue) {
+        LOGGER.debug("Type of amqpValue object  of received message " + messageId + " is " + amqpValue.getClass());
+        Class<?> clazz = amqpValue.getClass();
+        switch (clazz.getSimpleName()) {
+            case "Integer":
```

Comment:
What abt a `if else` construct for this? Does it make sense?

Reply:
as there can be more ASB value types (like the qpid binary datatype we found during the EI testing), kept the room for such cases. also I felt that switch clause provides a better readability and categorization for this usecase. WDYT?

Reply:
Should be fine if there's more cases.